core/src/main/java/org/mule/runtime/core/DefaultMuleMessage.java

```diff
@@ -124,17 +127,19 @@
         }
         else
         {
-            DataType<?> dataType = DataTypeFactory.create(payload.getClass(), previous.getDataType().getMimeType());
-            dataType.setEncoding(previous.getDataType().getEncoding());
+            DataType<?> dataType = DataTypeFactory.create(payload.getClass(),
+                    previous.getDataType().getMimeType(),
```

Comment:
Wouldn't it be best to add copy (template) functionality to factory and unit test it there?